### PR TITLE
Adding endpoint to order attachments

### PR DESCRIPTION
**Pull request by angela4713**
Adding endpoint to order attachments for a specific property. 

Documentation of from Immobilienscout [here](https://api.immobilienscout24.de/our-apis/import-export/attachments/attachmentsorder-put.html)

### CODE REVIEW

File: lib/immobilienscout/api/attachment.rb [hunk 1]
```diff
@@ -15,6 +15,16 @@ def add(is24_id, binary_file, metadata)
           parsed_response
         end
 
+        def order(is24_id, params)
```
Comment (jethroo) [inline lib/immobilienscout/api/attachment.rb]: would name it `put_order`  to make clear that one is **changing** the order.  If it just reads `order` one might easily think one would just get back the order of attachments.

File: lib/immobilienscout/api/attachment.rb [hunk 2]
```diff
@@ -15,6 +15,16 @@ def add(is24_id, binary_file, metadata)
...
+          raise ArgumentError unless params.present?
+
+          order_url = order_url(is24_id)
```
Comment (jethroo) [inline lib/immobilienscout/api/attachment.rb]: no need to introduce this local you can just go straight with 
```
parsed_response = Immobilienscout::Request.new(order_url(is24_id), params).put
```

File: lib/immobilienscout/api/attachment.rb [hunk 3]
```diff
@@ -25,6 +35,10 @@ def create_metadata_file(params)
         def add_url(is24_id)
           "#{Immobilienscout::Client.api_url}/restapi/api/offer/v1.0/user/me/realestate/#{is24_id}/attachment"
         end
+
+        def order_url(is24_id)
+          "#{Immobilienscout::Client.api_url}/restapi/api/offer/v1.0/user/me/realestate/#{is24_id}/attachment/attachmentsorder"
```
Comment (jethroo) [inline lib/immobilienscout/api/attachment.rb]: might not be done in this PR but I see this part dublicated quite often: `/restapi/api/offer/v1.0/user/me/realestate/`
also for the other endpoints so introducing a constant might make sense (imagine we would switch to v2 then all of those urls have to be edited)

File: spec/attachment_spec.rb [hunk 4]
```diff
@@ -37,4 +37,55 @@
...
+              '896939751',
+              '896939749',
+              '896939745'
+            ]
+          }
+        }
+      end
+
+      it 'returns success' do
+        VCR.use_cassette('attachments_are_successfuly_ordered_is24') do
```
Comment (wjfrey) [inline spec/attachment_spec.rb]: successfully 🤓

File: lib/immobilienscout/api/attachment.rb (hunk 3 above)
Comment (angela4713) [reply lib/immobilienscout/api/attachment.rb]: Yes good point, so far we just have the main part of the URL as a constant, but the rest is always repeated too, maybe we can take care of this in a new pull request

File: lib/immobilienscout/request.rb [hunk 5]
```diff
@@ -48,6 +48,19 @@ def post_with_multipart
...
+      headers = generate_headers(auth_header)
+
+      uri = URI.parse(@url)
+
+      request = Net::HTTP::Put.new(uri)
+      request.body = @params.to_json
+      request.initialize_http_header(headers)
+
+      execute_request(uri, request)
+    end
```
Comment (wjfrey) [inline lib/immobilienscout/request.rb]: I feel like we are repeating the same code in each action. But not hundred percent, I know 😅 - but maybe we can find a way to generalize at some point.

Comment (angela4713) [reply lib/immobilienscout/request.rb]: It would be a big refactor work! 😄

Comment (wjfrey) [reply lib/immobilienscout/request.rb]: Yes, that's why I added the "at some point"... 🙈

**Pull request merged**